store/tikv/lock_resolver.go

```diff
@@ -199,17 +200,32 @@ func (lr *LockResolver) BatchResolveLocks(bo *Backoffer, locks []*Lock, loc Regi
 		return false, nil
 	}
 
-	startTime := time.Now()
+	startTS, err := lr.store.GetOracle().GetTimestamp(bo.ctx)
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+
 	txnInfos := make(map[uint64]uint64)
+	startTime := time.Now()
 	for _, l := range expiredLocks {
 		if _, ok := txnInfos[l.TxnID]; ok {
 			continue
 		}
 
-		status, err := lr.getTxnStatus(bo, l.TxnID, l.Primary, 0)
+		currentTS, err := lr.store.GetOracle().GetLowResolutionTimestamp(bo.ctx)
 		if err != nil {
-			return false, errors.Trace(err)
+			return false, err
+		}
+		status, err := lr.getTxnStatus(bo, l.TxnID, l.Primary, startTS, currentTS)
+		if err != nil {
+			return false, err
+		}
+
+		if status.ttl > 0 {
+			// Do not clean lock that is not expired.
+			continue
```

Comment:
This function need to resolve all locks in `locks`. If some of the locks can't be resolved, this function shouldn't return `true`.

Reply:
Then large txn would block GC...

Reply:
To continue GC anyway, you should resolve **all** the locks before cleaning up the data. Otherwise not-committed secondaries may miss their primary.

Reply:
You may remove these locks regardless of their TTL, but you can never leave these locks there.

Reply:
We may need to update the CmdGCRequest instead of simply leave the lock here.
@coocood

Reply:
How will GC's logic be? By the way, GC Request is not used by default now.